packages/core/src/builder.ts

```diff
@@ -611,4 +617,46 @@ export default class SchemaBuilder<Types extends SchemaTypes> {
       ? processedSchema
       : lexicographicSortSchema(processedSchema);
   }
+
+  /**
+   * Helper for allowing plugins to fulfill the return of the `next` resolver, without paying the cost of the
+   * Promise if not required.
+   */
+  completeValue<T, R>(
+    valOrPromise: PromiseLike<T> | T,
+    onSuccess: (completedVal: T) => R,
+  ): R | Awaited<R> | Promise<Awaited<R>> | Promise<Awaited<R>>;
+
+  /**
+   * Helper for allowing plugins to fulfill the return of the `next` resolver, without paying the cost of the
+   * Promise if not required.
+   */
+  completeValue<T, R, E>(
+    valOrPromise: PromiseLike<T> | T,
+    onSuccess: (completedVal: T) => R,
+    onError: (err: unknown) => E,
+  ): R | Awaited<R> | Promise<Awaited<R>> | Promise<Awaited<R>>;
```

Comment:
Duplicate in the return type

Reply:
Also is Awaited<R> an actual possible return type?

Reply:
I think the `Awaited` was only needed as a return type when supporting the `PromiseLike` unwrapping thing I mentioned you had a question about above. It isn't a realistic scenario to support I don't think